adapters/handlers/rest/clusterapi/object_ttl.go

```diff
@@ -102,34 +111,58 @@ func (d *ObjectTTL) incomingDelete() http.Handler {
 			return
 		}
 
-		d.logger.WithFields(logrus.Fields{"action": "incoming_delete_expired_objects", "classes": len(body)}).Info("received request to delete expired objects")
+		d.logger.WithFields(logrus.Fields{
+			"action":  "object_ttl_deletions",
+			"classes": len(body),
+		}).Info("received request to delete expired objects")
 
 		// run the deletion in a separate goroutine to free up the HTTP handler immediately
 		enterrors.GoWrapper(func() {
 			// make sure to unlock the requestRunning flag when all deletions are done
 			defer d.requestRunning.Store(false)
+
+			start := time.Now()
+
+			ec := errorcompounder.NewSafe()
 			eg := enterrors.NewErrorGroupWrapper(d.logger)
+			eg.SetLimit(concurrency.TimesFloatGOMAXPROCS(d.config.ObjectsTTLConcurrencyFactor))
 
-			ec := errorcompounder.New()
-			for _, classPayload := range body {
-				className := classPayload.Class
+			objectsDeleted := atomic.Int32{}
+			onObjectsDeleted := func(count int32) { objectsDeleted.Add(count) }
 
-				idx, err := d.remoteIndex.IndexForIncomingWrite(context.Background(), className, 0)
-				if err != nil {
-					ec.Add(fmt.Errorf("get index for class %q: %w", className, err))
-					continue
+			for _, classPayload := range body {
+				collection := classPayload.Class
+				onCollectionError := func(err error) {
+					if err != nil {
+						ec.Add(fmt.Errorf("collection %q [%w]", collection, err))
+					}
 				}
+				// TODO aliszka:ttl add server shutdown context check?
 
-				err = idx.IncomingDeleteObjectsExpired(eg, classPayload.Prop, time.UnixMilli(classPayload.TtlMilli), time.UnixMilli(classPayload.DelMilli), 0)
+				// TODO aliszka:ttl index protected from closing/deletion?
+				// TODO aliszka:ttl schema version?
```

Comment:
I think this makes sure that the RAFT followers have the latest schemaversion to get the latest class definitions.

I think we could:
- continue using 0 - this means that the follower is using whatever verion it has locally. I _think_ this is ok, but not 100% sure if there is an edge case. As far as I can see, we are only using class settings here that are immutable (is MT etc)
- have the RAFT leader send its schema version (which always is the latest) with the request. Then the deletion coordinator might have to wait a little bit before it starts the delete. But that would also be fine, as object TTL deletions are rarely time-critical